invenio_search/api.py

```diff
@@ -117,3 +119,48 @@ def search(self):
                 return search_.response_class(partial(FacetedResponse, self))
 
         return RecordsFacetedSearch(query=query, filters=filters or {})
+
+    def with_preference_param(self):
+        """Add the preference param to the ES request and return a new Search.
+
+        The preference param avoids the bouncing effect with multiple
+        replicas, documented on ES documentation.
+        See: https://www.elastic.co/guide/en/elasticsearch/guide/current
+        /_search_options.html#_preference for more information.
+        """
+        user_hash = self._get_user_hash()
+        if user_hash:
+            return self.params(preference=user_hash)
+        return self
+
+    def _get_user_agent(self):
+        """Retrieve the request's User-Agent, if available.
+
+        Taken from Flask Login utils.py.
+        """
+        user_agent = request.headers.get('User-Agent')
+        if user_agent is not None:
+            user_agent = user_agent.encode('utf-8')
+        return user_agent or ''
+
+    def _get_remote_addr(self):
+        """Retrieve the request's address ip, if available.
+
+        Taken from Flask Login utils.py.
+        """
+        address = request.headers.get('X-Forwarded-For', request.remote_addr)
+        if address is not None:
+            # An 'X-Forwarded-For' header includes a comma separated list of
+            # the addresses, the first address being the actual remote address.
+            address = address.encode('utf-8').split(b',')[0].strip()
+        return address or ''
+
+    def _get_user_hash(self):
+        """Calculate a digest based on request's User-Agent and ip address."""
+        if request:
+            user_hash = '{ip}-{ua}'.format(ip=self._get_remote_addr(),
+                                           ua=self._get_user_agent())
+            alg = hashlib.md5()
+            alg.update(user_hash.encode('utf8'))
+            return alg.hexdigest()
+        return None
```

Comment:
IMHO we don't need ``return None`` is ``None`` by default :)

Reply:
Saying it explicitly makes the code more readable. Otherwise it might look like a forgotten return.